From 27ddb651cce225c17faff6391a10220df7fdf9c2 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Thu, 14 Jun 2018 11:47:00 -0700
Subject: [PATCH 2/2] Use gdm-password derived PAM config instead

By default, auth/account/password/session groups use the "other"
service, which will warn/deny access. So only specifying the auth group
is not sufficient to enable authentication on Clear Linux.

Instead, use a PAM config derived from the LFS gdm-password config as
installed by the gdm package.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 pam/i3lock | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/pam/i3lock b/pam/i3lock
index 6eb8e68..8b80576 100644
--- a/pam/i3lock
+++ b/pam/i3lock
@@ -2,5 +2,15 @@
 # PAM configuration file for the i3lock screen locker. By default, it includes
 # the 'login' configuration file (see /etc/pam.d/login)
 #
+# For Clear Linux OS, it is derived from the LFS gdm-password.pam in gdm.
 
-auth include login
+auth     requisite      pam_nologin.so
+auth     required       pam_env.so
+auth     required       pam_succeed_if.so uid >= 1000 quiet
+auth     include        system-auth
+
+account  include        common-account
+password include        common-password
+
+session  required       pam_limits.so
+session  include        common-session
-- 
2.19.1

